src/btree/bt_cursor.c

```diff
@@ -703,11 +793,8 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
      * exit.
      */
     if (btree->type == BTREE_ROW && WT_CURSOR_BOUNDS_SET(cursor)) {
-        WT_ERR(__wt_btcur_bounds_early_exit(session, cbt, true, &key_out_of_bounds));
-
-        if (!key_out_of_bounds)
-            WT_ERR(__wt_btcur_bounds_early_exit(session, cbt, false, &key_out_of_bounds));
-
+        WT_ERR(
```

Comment:
This is a new change, I updated the search code to use the new contains key function.